addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -369,39 +383,29 @@ var ImageWidget = MediaWidget.extend({
                 fields: ['name', 'datas_fname', 'mimetype', 'checksum', 'url', 'type', 'res_id', 'res_model', 'access_token'],
                 order: [{name: 'id', asc: false}],
                 context: weContext.get(),
-            },
+                // Try to fetch first record of next page just to know whether there is a next page.
+                limit: this.IMAGES_PER_PAGE + 1,
+                offset: pageNum * this.IMAGES_PER_PAGE,
+            }
         }).then(function (records) {
-            self.records = _.chain(records)
-                .filter(function (r) {
-                    return (r.type === "binary" || r.url && r.url.length > 0);
-                })
-                .uniq(function (r) {
-                    return (r.url || r.id);
-                })
-                .sortBy(function (r) {
-                    if (_.any(self.firstFilters, function (filter) {
-                        var regex = new RegExp(filter, 'i');
-                        return r.name.match(regex) || r.datas_fname && r.datas_fname.match(regex);
-                    })) {
-                        return -1;
-                    }
-                    if (_.any(self.lastFilters, function (filter) {
-                        var regex = new RegExp(filter, 'i');
-                        return r.name.match(regex) || r.datas_fname && r.datas_fname.match(regex);
-                    })) {
-                        return 1;
-                    }
-                    return 0;
-                })
-                .value();
-
+            self.lastLoadedPage = pageNum;
+            if (typeof pageNum !== 'number') {
```

Comment:
I remember we had to do this if in 11.0 but it seems like it's not needed anymore for this 11.3 patch? Or I am missing something?

Reply:
That is correct. Removed.